### PR TITLE
NEO-2457  Allow icon next to the label in both TextInput and Switch react components

### PR DESCRIPTION
**Pull request by yangAtSpoken**
[TextInput Label with Icon](https://deploy-preview-572--neo-react-library-storybook.netlify.app/?path=/story/components-text-input--label-with-icon)
[Switches Label with Icon, scroll the bottom of the story](https://deploy-preview-572--neo-react-library-storybook.netlify.app/?path=/story/components-switch--default)
[Figma reference: TextInput](https://www.figma.com/design/SQHQlIhFHWCjngHZOnK8MH/Web-Components-2.0?m=auto&node-id=1-144&t=VeWAoZcNyrp0nsA6-1)
[Figma reference: Label](https://www.figma.com/design/SQHQlIhFHWCjngHZOnK8MH/Web-Components-2.0?m=auto&node-id=907-311198&t=VeWAoZcNyrp0nsA6-1)
[Figma reference: Kin Ye](https://www.figma.com/design/n1vWAQ0pgib4s7M1uig987/05---NGM---Features---Contact-Center?node-id=1865-112249&node-type=canvas&t=xRu03FCUulgxYdLQ-0)

**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the Deploy Preview link at the top of this PR (or remove it if not applicable)
- [x] updated the Figma referense link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [x] tagged `@avaya-dux/dux-design` if any visual changes have occurred
- [x] tagged `@avaya-dux/dux-devs`

`@avaya-dux/dux-design`: 
- please check both visuals and keyboard interaction with the Icon in both TextInput and Switch stories lined above;

`@avaya-dux/dux-devs`: 
- created an internal Label component, not exported externally as no use cases yet;
- support both TextInput and Switch component;
- allow passing custom classes to TextInput;

Screenshot 2024-10-23 at 11 25 50 AM
Screenshot 2024-10-23 at 11 25 34 AM

## Release Notes

- **New Features**
	- Introduced a `Label` component for enhanced label rendering with optional icons.
	- Added `labelIcon` prop to the `Switch` component for icon integration in labels.
	- New `LabelWithIcon` story for the `TextInput` component showcasing labeled icons.
	- Created `IconTooltip` story for the `Tooltip` component with an icon.

- **Bug Fixes**
	- Improved accessibility compliance in the `LabelWithIcon` tests.

- **Documentation**
	- Enhanced Storybook documentation for `Switch`, `TextInput`, and `Tooltip` components with new examples.

- **Style**
	- Added new CSS styles for labels and text inputs to improve layout and visual structure.

- **Chores**
	- Updated version number in `package.json` from `1.4.1` to `1.4.2`.

### CODE REVIEW

File: src/components/TextInput/TextInput.stories.tsx [hunk 1]
```diff
@@ -22,6 +23,46 @@ export const Default = () => {
...
+				isSmall
+				placeholder="input: small, label: icon"
+			/>
+
+			{/* custom rule */}
+			<div className="container--narrow">
+				<TextInput
+					label={label}
+					required
+					className="custom-rule--narrow"
```
Comment (yangAtSpoken) [inline src/components/TextInput/TextInput.stories.tsx:59]: Passing a user provided class to TextInput

File: src/components/Label/Label.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+			{justLabel}
+			<Tooltip label={icon.iconTooltipText} position={icon.iconTooltipPosition}>
+				<Icon
+					tabIndex={0}
+					icon={icon.iconType}
+					aria-label={icon.iconAriaLabel}
+					size="sm"
+				/>
+			</Tooltip>
+		</div>
```
[the remark this answers is not in the document]
Comment (yangAtSpoken) [reply src/components/Label/Label.tsx:50]: Tooltip opens automatically when icon receives keyboard focus.  No need to react to other key presses.

File: src/components/Switch/Switch.stories.tsx [hunk 3]
```diff
@@ -21,47 +22,69 @@ const DirectionTemplate: Story<DirectionTemplateProps> = ({
 	const longText =
 		"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean rhoncus non elit eu mollis.";
 
+	const labelIcon = {
+		iconType: "info",
+		iconAriaLabel: "Info",
+		iconTooltipPosition: "auto",
+		iconTooltipText: "This is a tooltip",
+	} as LabelIconProps;
```
[the remark this answers is not in the document]
Comment (yangAtSpoken) [reply src/components/Switch/Switch.stories.tsx:30]: No need to test Tooltip here.

File: src/components/TextInput/TextInput.stories_shim.css [hunk 4]
```diff
@@ -0,0 +1,7 @@
+.container--narrow {
+	max-width: 280px;
+}
+.custom-rule--narrow {
+	color: orange;
```
Comment (joe-s-avaya) [inline src/components/TextInput/TextInput.stories_shim.css:5]: would prefer that we use neo colors, but as these are story colors, I won't enforce that rule

File: src/components/TextInput/TextInput.tsx [hunk 5]
```diff
@@ -225,18 +226,19 @@ export const InternalTextInputElement = ({
 	readOnly,
 	type,
 	value,
+	className,
```
Comment (joe-s-avaya) [inline src/components/TextInput/TextInput.tsx:229]: excellent, thanks for taking care of that 👍

File: src/components/Label/Label_shim.css [hunk 6]
```diff
@@ -0,0 +1,60 @@
+.neo-input-group .neo-form__label-with-icon {
+	display: flex;
+	align-items: center;
+	gap: 2px;
+	padding-bottom: 4px;
+
+	/* TextInput label */
+	label {
```
Comment (razianmm) [inline src/components/Label/Label_shim.css:8]: I might need a refresher on how these shim files are scoped, but is there a potential for this element-level scoping in the CSS to cause issues with other HTML `label` elements on the page?

**Review by joe-s-avaya — APPROVED** [review]
This is good, **_BUT_** we _must_ have all CSS in neo-css so that all teams (especially Miruna's) get access to all the things.

I presume that you are putting the label shim in this PR so that we can get quick feedback from Matt+Justin. Please ensure that you move all the CSS our of neo-react and into neo-css.

**Review by razianmm — APPROVED** [review]
One question re: CSS scoping but otherwise looks great to me 👍

File: src/components/Label/Label_shim.css (hunk 6 above)
Comment (joe-s-avaya) [reply src/components/Label/Label_shim.css:8]: shim files are not scoped, Mo is right, this will need increased specificity

Comment (razianmm) [reply src/components/Label/Label_shim.css:8]: Wait I see now; it's nested within the above class hierarchy. Disregard 🙏

**Comment by yangAtSpoken** [comment]
> This is good, **_BUT_** we _must_ have all CSS in neo-css so that all teams (especially Miruna's) get access to all the things.
> 
> I presume that you are putting the label shim in this PR so that we can get quick feedback from Matt+Justin. Please ensure that you move all the CSS our of neo-react and into neo-css.

Will have a second PR to move shim file to neo css.

**Comment by matthewAvaya** [comment]
Overall, this is looking good. The VO sounds different on the Text Input compared to the Switch. Not sure if it's a big deal or not. Is there a way to share these links with Terri or does she need access permissions?

**Comment by yangAtSpoken** [comment]
> Overall, this is looking good. The VO sounds different on the Text Input compared to the Switch. Not sure if it's a big deal or not. Is there a way to share these links with Terri or does she need access permissions?

You could share the first two preview link with her. I believe those links are open to the world.

**Comment by Justin-Avaya** [comment]
Looks good, but the spacing between the * and the info icon is slightly off. 

example of what is in the deploy preview
Screenshot 2024-10-23 at 2 11 12 PM

example of the correct spacing (2px spacing between elements)
Screenshot 2024-10-23 at 10 29 00 AM

**Review by enrique-prado — APPROVED** [review]

**Comment by Justin-Avaya** [comment]
@yangAtSpoken - late breakin design change... Neo Design would like to make the following changes to the label.
Thanks.

Screenshot 2024-10-23 at 5 35 52 PM

**Comment by yangAtSpoken** [comment]
@Justin-Avaya updated to the new design. Please look again.

**Review by Justin-Avaya — APPROVED** [review]
Looks great. Nice work.

**Review by matthewAvaya — APPROVED** [review]

**Pull request merged**